From b38a4fdfd631a800d4171a4eb3e519075755f430 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Tue, 13 Jun 2017 15:23:58 +0100
Subject: [PATCH] Fix a crash when refreshing metadata

When processing the file handles it seems g_variant_get() actually wants to
write the fd integer. This bug does not happen every time, and seems to vanish
every time under valgrind.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1460429
---
 src/fu-main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/fu-main.c b/src/fu-main.c
index f1fdd68..f9b1973 100644
--- a/src/fu-main.c
+++ b/src/fu-main.c
@@ -2087,8 +2087,8 @@ fu_main_daemon_method_call (GDBusConnection *connection, const gchar *sender,
 		gint fd_data;
 		gint fd_sig;
 
-		g_variant_get (parameters, "(&shh)", &id);
-		g_debug ("Called %s(%s)", method_name, id);
+		g_variant_get (parameters, "(&shh)", &id, &fd_data, &fd_sig);
+		g_debug ("Called %s(%s,%i,%i)", method_name, id, fd_data, fd_sig);
 
 		message = g_dbus_method_invocation_get_message (invocation);
 		fd_list = g_dbus_message_get_unix_fd_list (message);
